build/_k-test.shade

```diff
@@ -3,6 +3,7 @@ use import="Environment"
 
 default NO_PARALLEL_TEST_PROJECTS='${E("NO_PARALLEL_TEST_PROJECTS")}'
 default KOREBUILD_TEST_DNXCORE='${E("KOREBUILD_TEST_DNXCORE")}'
+default CORECLR_RUNTIME_VERSION='${E("CORECLR_RUNTIME_VERSION")}'
```

Comment:
CLR is tested correctly since it sets %PATH% before invoking this method. Since for Clr we do just `dnx test` it runs using the correct runtime. We do the same for CoreClr but this method happily overwrites the setting running against `default`. 
Another option would be to do something similar in this method - get the path of the current runtime and change it to coreclr if it is not already coreclr and run without dnvm which would solve both issues.